### PR TITLE
Update armcore with latest azcore

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] Apache v2 license headers are included in each file.
 
[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/master/CHANGELOG.md

### CODE REVIEW

File: sdk/armcore/poller_test.go [hunk 1]
```diff
@@ -27,10 +27,8 @@ type mockType struct {
 func getPipeline(srv *mock.Server) azcore.Pipeline {
 	return azcore.NewPipeline(
 		srv,
-		azcore.NewTelemetryPolicy(azcore.TelemetryOptions{}),
```
Comment (catalinaperalta) [inline sdk/armcore/poller_test.go:30]: Just curious, why was this removed?

**Review by catalinaperalta — APPROVED** [review]
LGTM

File: sdk/armcore/poller_test.go (hunk 1 above)
Comment (jhendrixMSFT) [reply sdk/armcore/poller_test.go:30]: It's not germane to the test so I removed it.
